### DISCUSSION TITLE
Telegraf unknown version

### DISCUSSION THREAD

**Discussion by Hipska** [Homebrew/discussions#3710] [Tap maintenance and Homebrew development] [answered]
After installing latest version of `telegraf`, the tool doesn't know its version anymore.

Before:
```
$ telegraf --version
Telegraf 1.22.1
```

After `brew upgrade telegraf`:
```
$ telegraf --version
Telegraf unknown
```

According to the logs, version 1.24.1 should be installed. But looking at the formula, I have some questions/suggestions.

1. Is there a reason the formula install does a `go build` instead of the upstream recommended `make build` command?
2. The bottles seem to be compiled from this formula instead of the pre-compiled versions from upstream?

Both would resolve the current issue of telegraf being wrongly compiled and not knowing its own version number.

**Comment by SMillerDev** [comment, marked as answer]
> Is there a reason the formula install does a go build instead of the upstream recommended make build command?

Because there's not much difference between the two and it saves a dependency on make: https://github.com/influxdata/telegraf/blob/master/Makefile#L127

> The bottles seem to be compiled from this formula instead of the pre-compiled versions from upstream?

Nothing in Homebrew/core uses upstream binaries. Everything is build from source.

**Reply by Hipska** [reply]
Okay, lets deep dive into it, we are talking about this line: https://github.com/Homebrew/homebrew-core/blob/b0cdb314aa49c673203a23691d08c35ee461c92a/Formula/telegraf.rb#L26

1. Where is `#{version}` defined?
2. I see a lot differences to the used `-ldflags` compared to when I run make on the repo: `go build -tags "" -ldflags " -X github.com/influxdata/telegraf/internal.Commit=94e39fa0 -X github.com/influxdata/telegraf/internal.Branch=master -X github.com/influxdata/telegraf/internal.Version=1.25.0-94e39fa0" ./cmd/telegraf`

**Reply by SMillerDev** [reply]
Version is extracted from the URL. And if the tags changed upstream they should be changed in the formula too.

**Reply by Hipska** [reply]
Well yeah that's exactly my point. There IS a difference on how homebrew builds the tool vs upstream builds the tool..

**Reply by Hipska** [reply]
Created a PR to fix: https://github.com/Homebrew/homebrew-core/pull/113113

**Reply by Hipska** [reply]
However, I would need some help on how to define the Branch and Commit flag when building/installing head..
